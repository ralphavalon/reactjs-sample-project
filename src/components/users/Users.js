import React from 'react';
import './Users.css';

function Users(props) {
  return (
    <div className="App">
      <header className="App-header">
        <ul>
            {props.users.map((user) => (
                <li key={user.id}>{user.name} - {user.age} - {user.city}</li>
            ))}
        </ul>
      </header>
    </div>
  );
}

export default Users;
