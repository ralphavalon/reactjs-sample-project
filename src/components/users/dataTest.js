export const users = [{
    id: 1,
    name: 'Ralph Avalon',
    city: 'Rio de Janeiro',
    age: 13
}];
