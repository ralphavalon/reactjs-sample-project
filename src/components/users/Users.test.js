import React from 'react';
import { render } from '@testing-library/react';
import Users from './Users';
import * as dataTest from './dataTest';

test('renders empty user list', () => {
  const { container } = render(<Users users={[]} />);
  expect(container).toMatchSnapshot();
});

test('renders user list', () => {
  const { getByText, container } = render(<Users users={dataTest.users} />);
  const linkElement = getByText(/Ralph Avalon/i);
  expect(linkElement).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
