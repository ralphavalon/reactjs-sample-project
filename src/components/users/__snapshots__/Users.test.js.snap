// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty user list 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <ul />
    </header>
  </div>
</div>
`;

exports[`renders user list 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <ul>
        <li>
          Ralph Avalon
           - 
          13
           - 
          Rio de Janeiro
        </li>
      </ul>
    </header>
  </div>
</div>
`;
